File: beecrowd_exercicios_em_Cpp/beginner/triangle_types.cpp
#include <iostream>

using namespace std;

void ordenar(double arr_ord []){

    int ok {0};

     while (!ok){
        ok = 1;
        for(int i = 0; i < 2; i++){
            if (arr_ord[i] < arr_ord[i + 1]){
                ok = 0;
                double temp = arr_ord[i];
                arr_ord[i] = arr_ord[i+1];
                arr_ord[i+1] = temp;
            }
        }
    }

}

void triangle_types() {
    double sides[3];

    for (int i = 0; i < 3; i++) cin >> sides[i];

    ordenar(sides);

    if(sides[0] < sides[1]+sides[2] && sides[1] < sides[0]+sides[2] && sides[2] < sides[0]+sides[1]){

        if (sides[0] * sides[0] == sides[1] * sides[1] + sides[2] * sides[2]) {
            cout << "TRIANGULO RETANGULO\n";
            }

        if (sides[0] * sides[0] > sides[1] * sides[1] + sides[2] * sides[2]) {
            cout << "TRIANGULO OBTUSANGULO\n";
            }

        if (sides[0] * sides[0] < sides[1] * sides[1] + sides[2] * sides[2]) {
            cout << "TRIANGULO ACUTANGULO\n";
            }

        if (sides[0] == sides[1] && sides[0] == sides[2] && sides[1] == sides[2]) {
            cout << "TRIANGULO EQUILATERO\n";
            }

        if (sides[0] == sides[1] and sides[1] != sides[2] || sides[1] == sides[2] and sides[0] != sides[2]
        || sides[2] == sides[0] and sides[1] != sides[0]) {
            cout << "TRIANGULO ISOSCELES\n";
            }
        return;
    }
    cout << "NAO FORMA TRIANGULO\n";

}

int main(){

    triangle_types();

    return 0;
}

File: beecrowd_exercicios_em_python/begginner/triangle_types.py
def swap(d, e, f):
    d[e], d[f] = d[f], d[e]


def sort_sides(d, e, f):
    array = [d, e, f]

    for i in range(1, len(array)):
        j = i
        while j > 0 and array[j] > array[j - 1]:
            swap(array, j - 1, j)
            j -= 1

    return array


def get_types(t, u, v):
    values = {}
    triangles = [
        "RETANGULO", "OBTUSANGULO",
        "ACUTANGULO", "EQUILATERO",
        "ISOSCELES"
    ]

    for i in range(len(triangles)):
        values["TRIANGULO " + triangles[i]] = \
            check_types(i, t, u, v)

    return values


def check_types(s, t, u, v):
    checks = [t ** 2 == u ** 2 + v ** 2,
              t ** 2 > u ** 2 + v ** 2,
              t ** 2 < u ** 2 + v ** 2,
              t == u == v,
              t == u != v or u == v != t or v == t != u
              ]

    return checks[s]


def show_results(w):
    for k in w.keys():
        if w[k]:
            print(k)


def triangle_types(x, y, z):
    sides = sort_sides(x, y, z)

    if sides[0] >= sides[1] + sides[2]:
        print("NAO FORMA TRIANGULO")
        return

    types = get_types(sides[0], sides[1], sides[2])

    show_results(types)


a, b, c = map(float, input().split())

triangle_types(a, b, c)
